demo/src/components/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import { FormEvent, useContext, useRef } from 'react';
+import cn from 'classnames';
+
+import { SpacesContext } from '.';
+import { Member } from '../utils/types';
+import { getRandomColor } from '../utils';
+
+interface Props {
+  self?: Member;
+  isVisible?: boolean;
+  setIsVisible?: (isVisible: boolean) => void;
+}
+
+export const Modal = ({ isVisible = false, setIsVisible, self }: Props) => {
+  const space = useContext(SpacesContext);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    if (!space || !setIsVisible) return;
+
+    space.updateProfileData({ name: inputRef.current?.value, color: getRandomColor() });
```

Comment:
passing funciton as an argument is still not working, `currentProfile` is `null` there, so to pass the `isMember` check in `useMembers` hook, I'm regenerating the color here. Not a big deal, but also, not ideal, imo.